agent/handlers/agentapi/taskprotection/v1/handlers/handlers.go

```diff
@@ -137,90 +146,46 @@ func UpdateTaskProtectionHandler(state dockerstate.TaskEngineState, credentialsM
 
 		// there are no exceptions but there are failures when setting protection in scheduler
 		if len(response.Failures) > 0 {
-			writeJSONResponse(w, http.StatusBadRequest, response.Failures[0], updateTaskProtectionRequestType)
+			if len(response.Failures) != ExpectedProtectionResponseLength {
```

Comment:
Should we use `>` here? 

`>` and `!=` are the same if `ExpectedProtectionResponseLength=1`, but since we are using it as a variable, we should try to use the comparison (especially when the log message also says "expect at most ... failure")

Reply:
ok